magneticfield/inc/Geant/ScalarFieldTrack.h

```diff
@@ -0,0 +1,338 @@
+      // fPolarization( rStVec.fPolarization ),
+      // fMomentumDir( rStVec.fMomentumDir ),
+      fCharge(rStVec.fCharge)
+{
+  SixVector[0] = rStVec.SixVector[0];
+  SixVector[1] = rStVec.SixVector[1];
+  SixVector[2] = rStVec.SixVector[2];
+  SixVector[3] = rStVec.SixVector[3];
+  SixVector[4] = rStVec.SixVector[4];
+  SixVector[5] = rStVec.SixVector[5];
+
+  // fpChargeState= new G4ChargeState( *rStVec.fpChargeState );
+  // Can share charge state only when using handles etc
+  //   fpChargeState = rStVec.fpChargeState;
+}
+
+inline ScalarFieldTrack::~ScalarFieldTrack()
+{
+  // delete fpChargeState;
+}
+
+inline void ScalarFieldTrack::SetCurvePnt(const ThreeVector &pPosition, const ThreeVector &pMomentum, double s_curve)
+{
+  SixVector[0] = pPosition.x();
+  SixVector[1] = pPosition.y();
+  SixVector[2] = pPosition.z();
+
+  SixVector[3] = pMomentum.x();
+  SixVector[4] = pMomentum.y();
+  SixVector[5] = pMomentum.z();
+
+  fMomentumMag = pMomentum.Mag();
+  if (fMomentumMag > 0.0) {
+    // fMomentumDir = (1.0/fMomentumMag) * pMomentum;
+  }
+  fDistanceAlongCurve = s_curve;
+
+  // return *this;
+}
+
+inline ThreeVector ScalarFieldTrack::GetPosition() const
+{
+  ThreeVector myPosition(SixVector[0], SixVector[1], SixVector[2]);
+  return myPosition;
+}
+
+inline ThreeVector ScalarFieldTrack::GetMomentumDirection() const
+{
+  double inv_mag = 1.0 / fMomentumMag;
+  return inv_mag * ThreeVector(SixVector[3], SixVector[4], SixVector[5]);
+}
+
+inline void ScalarFieldTrack::SetPosition(const ThreeVector &pPosition)
+{
+  SixVector[0] = pPosition.x();
+  SixVector[1] = pPosition.y();
+  SixVector[2] = pPosition.z();
+}
+
+inline void ScalarFieldTrack::SetMomentum(const ThreeVector &vMomentum)
+{
+  SixVector[3] = vMomentum.x();
+  SixVector[4] = vMomentum.y();
+  SixVector[5] = vMomentum.z();
+  fMomentumMag = vMomentum.Mag();
+}
+
+inline double ScalarFieldTrack::GetMomentumMag() const
+{
+  return fMomentumMag;
+}
+
+inline double ScalarFieldTrack::GetCurveLength() const
+{
+  return fDistanceAlongCurve;
+}
+
+inline void ScalarFieldTrack::SetCurveLength(double nCurve_s)
+{
+  fDistanceAlongCurve = nCurve_s;
+}
+
+// inline double ScalarFieldTrack::GetKineticEnergy() const
+// { return fKineticEnergy; }
+
+// inline void ScalarFieldTrack::SetKineticEnergy(double newKinEnergy)
+// {  fKineticEnergy=newKinEnergy; }
+
+// inline ThreeVector ScalarFieldTrack::GetPolarization() const
+// { return fPolarization; }
+
+// inline void ScalarFieldTrack::SetPolarization(const ThreeVector& vecPlz)
+// { fPolarization= vecPlz; }
+
+#if 0
+inline
+double ScalarFieldTrack::GetLabTimeOfFlight() const
+{
+   return fLabTimeOfFlight;
+}
+
+inline
+void ScalarFieldTrack::SetLabTimeOfFlight(double nTOF)
+{
+   fLabTimeOfFlight=nTOF;
+}
+
+inline
+double  ScalarFieldTrack::GetProperTimeOfFlight() const
+{
+   return fProperTimeOfFlight;
+}
+
+inline
+void ScalarFieldTrack::SetProperTimeOfFlight(double nTOF)
+{
+   fProperTimeOfFlight=nTOF;
+}
+#endif
```

Comment:
Should dead code be removed from the skeleton for VectorFlow? It would look cleaner. Like this it's confusing for the student.

Reply:
Yes, I will do this later